src/mgmt/srv_util.c

```diff
@@ -80,3 +80,17 @@ ds_mgmt_group_update(crt_group_mod_op_t op, struct server_entry *servers,
 		d_rank_list_free(ranks);
 	return rc;
 }
+
+void
+ds_mgmt_kill_rank(bool force)
+{
+	int	sig;
+
+	if (force)
+		sig = SIGKILL;
+	else
+		sig = SIGTERM;
+
+	D_INFO("Kill with signal %d\n", sig);
+	kill(getpid(), sig);
+}
```

Comment:
It seems a little weird to have this here... Why would we rely on dRPC to tell a rank to kill itself when we can just kill it directly from the control plane? What happens if the instance is so messed up that it can't handle this request?

Reply:
this is the original kill_rank code and I haven't done much with it, this has been separated out to enable mocking in the drpc handler tests. I guess it might be useful if we want to do some specific type of shutdown but we could do that in prep_shutdown. we can do this from the harness if that would be preferred (selecting which signal based on the force flag as in here). I think that having the option of triggering via drpc is not a bad thing so if change is requested I would prefer to doing a follow-up.

Reply:
Yeah, I think prep_shutdown logic should be distinct, and we shouldn't rely on being able to command a rank to kill itself. If you really want to do it as another PR, please create a ticket for this so that it's not lost. This is not at all a reliable way to do this.

Reply:
the stop --force option is already in place and I will work on what you request. https://jira.hpdd.intel.com/browse/DAOS-4018 has been updated with this requirement.